lib-ee/emqx_ee_bridge/src/gcp_pubsub/emqx_ee_bridge_gcp_pubsub_resource.erl

```diff
@@ -0,0 +1,599 @@
+    emqx_connector_jwt_sup:ensure_worker_deleted(JWTWorkerId),
+    ehttpc_sup:stop_pool(PoolName).
+
+-spec on_query(
+    bridge_id(),
+    {send_message, map()},
+    state()
+) ->
+    {ok, status_code(), headers()}
+    | {ok, status_code(), headers(), body()}
+    | {error, {recoverable_error, term()}}
+    | {error, term()}.
+on_query(BridgeId, {send_message, Selected}, State) ->
+    Requests = [{send_message, Selected}],
+    ?TRACE(
+        "QUERY_SYNC",
+        "gcp_pubsub_received",
+        #{requests => Requests, connector => BridgeId, state => State}
+    ),
+    do_send_requests_sync(State, Requests, BridgeId).
+
+-spec on_query_async(
+    bridge_id(),
+    {send_message, map()},
+    {ReplyFun :: function(), Args :: list()},
+    state()
+) -> ok.
+on_query_async(BridgeId, {send_message, Selected}, ReplyFunAndArgs, State) ->
+    Requests = [{send_message, Selected}],
+    ?TRACE(
+        "QUERY_ASYNC",
+        "gcp_pubsub_received",
+        #{requests => Requests, connector => BridgeId, state => State}
+    ),
+    do_send_requests_async(State, Requests, ReplyFunAndArgs, BridgeId).
+
+-spec on_batch_query(
+    bridge_id(),
+    [{send_message, map()}],
+    state()
+) ->
+    {ok, status_code(), headers()}
+    | {ok, status_code(), headers(), body()}
+    | {error, {recoverable_error, term()}}
+    | {error, term()}.
+on_batch_query(BridgeId, Requests, State) ->
+    ?TRACE(
+        "QUERY_SYNC",
+        "gcp_pubsub_received",
+        #{requests => Requests, connector => BridgeId, state => State}
+    ),
+    do_send_requests_sync(State, Requests, BridgeId).
+
+-spec on_batch_query_async(
+    bridge_id(),
+    [{send_message, map()}],
+    {ReplyFun :: function(), Args :: list()},
+    state()
+) -> ok.
+on_batch_query_async(BridgeId, Requests, ReplyFunAndArgs, State) ->
+    ?TRACE(
+        "QUERY_ASYNC",
+        "gcp_pubsub_received",
+        #{requests => Requests, connector => BridgeId, state => State}
+    ),
+    do_send_requests_async(State, Requests, ReplyFunAndArgs, BridgeId).
+
+-spec on_get_status(manager_id(), state()) -> connected | disconnected.
+on_get_status(InstanceId, State) ->
+    #{
+        connect_timeout := Timeout,
+        pool_name := PoolName
+    } = State,
+    case do_get_status(InstanceId, PoolName, Timeout) of
+        true ->
+            connected;
+        false ->
+            ?SLOG(error, #{
+                msg => "gcp_pubsub_bridge_get_status_failed",
+                state => State
+            }),
+            disconnected
+    end.
+
+%%-------------------------------------------------------------------------------------------------
+%% Helper fns
+%%-------------------------------------------------------------------------------------------------
+
+-spec ensure_jwt_worker(manager_id(), config()) ->
+    #{
+        jwt_worker_id := jwt_worker(),
+        project_id := binary()
+    }.
+ensure_jwt_worker(InstanceId, #{
+    service_account_json := ServiceAccountJSON,
```

Comment:
NIT(naming): `service_account_json` and `ServiceAccountJSON` seem to be maps and not strings, while JSON is a string representation 😅 I mean when we parse JSON it is no more JSON but just data 🤔